fs/eol.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+/** EndOfLine character enum */
+export enum EOL {
+  LF = "\n",
+  CRLF = "\r\n"
+}
+
+const regDetect = /(?:\r?\n)/g;
+const regCRLF = /\r\n/g;
+const regLF = /\n/g;
+
+/**
+ * Detect the EOL character for string input.
+ * returns null if no newline
+ */
+export function detect(content: string): EOL | null {
+  const d = content.match(regDetect);
+  if (!d || d.length === 0) {
+    return null;
+  }
+  const crlf = d.filter(x => x === EOL.CRLF);
+  if (crlf.length > 0) {
+    return EOL.CRLF;
+  } else {
+    return EOL.LF;
+  }
+}
+
+/** Format the file to the targeted EOL */
+export function format(content: string, eol: EOL): string {
+  const _eol = detect(content);
+  if (!_eol) {
+    return content;
+  } else if (_eol === eol) {
+    return content;
```

Comment:
This doesn’t work if the file has mixed newlines, since `detect` only looks at the first newline.

Reply:
I was really wondering about this case. Is that really possible? And in this case, what would you do? Because the file is somehow invalid no?